app/controllers/users_controller.rb

```diff
@@ -10,7 +10,7 @@ def show
     end
 
     def update
-        @user = current_user.users.find(params[:id])
+        @user = User.find(params[:id])
```

Comment:
Do you want to allow users to update eachother's profiles?  Seems like a security hole.  Why not just:

```
@user = current_user
```